src/aws-elasticache/serverless-cache.ts

```diff
@@ -0,0 +1,502 @@
+import {
+  IResource,
+  Lazy,
+  Names,
+  Resource,
+  Stack,
+  aws_elasticache,
+  aws_ec2,
+  Token,
+  aws_iam,
+  ArnFormat,
+} from 'aws-cdk-lib';
+import { IKey } from 'aws-cdk-lib/aws-kms';
+import { Construct } from 'constructs';
+import { DailySnapshotTime } from './daily-snapshot-time';
+import { IUserGroup } from './user-group';
+import { Engine } from './util';
+
+/**
+ * A ElastiCache Serverless Cache
+ */
+export interface IServerlessCache extends IResource, aws_ec2.IConnectable {
+  /**
+   * The serverless cache ARN.
+   *
+   * @attribute
+   */
+  readonly serverlessCacheArn: string;
+
+  /**
+   * The serverless cache name.
+   *
+   * @attribute
+   */
+  readonly serverlessCacheName: string;
+
+  /**
+   * The DNS hostname of the cache node.
+   *
+   * @attribute
+   */
+  readonly endpointAddress: string;
+
+  /**
+   * The port number that the cache engine is listening on.
+   *
+   * @attribute
+   */
+  readonly endpointPort: number;
+
+  /**
+   * Grant the given identity the specified actions
+   */
+  grant(grantee: aws_iam.IGrantable, ...actions: string[]): aws_iam.Grant;
+
+  /**
+   * Grant the given identity connection access to the cache.
+   */
+  grantConnect(grantee: aws_iam.IGrantable): aws_iam.Grant;
+}
+
+/**
+ * The version number of the engine the serverless cache is compatible with.
+ */
+export enum MajorVersion {
+  /**
+   * Version 7
+   */
+  VER_7 = '7',
+}
+
+/**
+ * Properties for defining a ElastiCache Serverless Cache.
+ */
+export interface ServerlessCacheProps {
+  /**
+   * The engine the serverless cache is compatible with.
+   */
+  readonly engine: Engine;
+
+  /**
+   * The unique identifier of the serverless cache.
+   * The name can have up to 40 characters, and must not contain spaces.
+   *
+   * @default - auto generate
+   */
+  readonly serverlessCacheName?: string;
+
+  /**
+   * The daily time when a cache snapshot will be created.
+   * This property must be set along with `snapshotRetentionLimit`.
+   *
+   * @default - ElastiCache automatically assigns the backup window if \`snapshotRetentionLimit\` is set. Otherwise, no snapshots are taken.
+   * @see https://docs.aws.amazon.com/AmazonElastiCache/latest/dg/backups-automatic.html
+   */
+  readonly dailySnapshotTime?: DailySnapshotTime;
+
+  /**
+   * A description of the serverless cache.
+   * The description can have up to 255 characters and must not contain < and > characters.
```

Comment:
This constraint is quoted from the management console. It is not written in the [CFn documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-elasticache-serverlesscache.html#cfn-elasticache-serverlesscache-description).

<img width="621" alt="image" src="https://github.com/user-attachments/assets/14292860-1aa5-4ecd-a791-af43e4a45ae1">